File: micronaut-java-parent-example/domain/src/main/java/io/archimedesfw/maven/micronaut/example/todo/persistence/TodoRepository.java
package io.archimedesfw.maven.micronaut.example.todo.persistence;

import io.archimedesfw.maven.micronaut.example.todo.Todo;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.repository.CrudRepository;

import static io.micronaut.data.model.query.builder.sql.Dialect.POSTGRES;

@JdbcRepository(dialect = POSTGRES)
public interface TodoRepository extends CrudRepository<Todo, Integer> {

}

File: micronaut-kotlin-parent-example/domain/src/main/kotlin/io/archimedesfw/maven/micronaut/example/todo/persistence/TodoRepository.kt
package io.archimedesfw.maven.micronaut.example.todo.persistence

import io.archimedesfw.maven.micronaut.example.todo.Todo
import io.micronaut.data.jdbc.annotation.JdbcRepository
import io.micronaut.data.model.query.builder.sql.Dialect.POSTGRES
import io.micronaut.data.repository.CrudRepository

@JdbcRepository(dialect = POSTGRES)
interface TodoRepository : CrudRepository<Todo, Int>
